### ISSUE TITLE
Output chunks change when build stays the same

### ISSUE THREAD

**Issue by chriskrogh** [vercel/next.js#70754]
### Link to the code that reproduces this issue

https://github.com/chriskrogh/build-manifest-test

### To Reproduce

1. clone the repo and run `npm i`
2. run the nextjs build (`npm run build`) and take note of the `app-build-manifest.json`
3. run the nextjs build again and compare the `app-build-manifest.json`s across the two builds 

* you may want to sort them by route if they're not in the same order across builds
* you may also need to repeat step 3 if you observe identical chunk names. see below for why that is

## Current
you'll notice that even though **the build hasn't changed**, webpack creates new chunks for some of the pages.

<img width="1139" alt="Screenshot 2024-10-03 at 11 41 29 AM" src="https://github.com/user-attachments/assets/11cc1161-67b3-42e8-9902-6068ffc4b9b0">

https://www.diffchecker.com/OidQJCx0/

according to [webpack](https://webpack.js.org/guides/caching/):
> The [contenthash] substitution will add a unique hash based on the content of an asset. When the asset's content changes, [contenthash] will also change.

if you compare the changed `page-[contenthash].js` chunks across builds, you'll notice that the contents are nearly identical, except for the webpack chunk IDs at the top of the file. the contents of the array are the same, it's just that the order can change across builds

<img width="1330" alt="Screenshot 2024-10-03 at 12 05 47 PM" src="https://github.com/user-attachments/assets/ca167455-eb34-4550-b63f-10ae00d3fba0">

https://www.diffchecker.com/Zp1tnr7X/

i.e. because the chunk IDs aren't sorted at the top of the chunk, webpack treats these as "new" chunks, even though the actual code hasn't changed.

### Why is this important?

* This can hurt site performance since browsers cache JS chunks based on the chunk name.
  * (fetching from the CDN is slower than not fetching from the CDN)
  * At Faire, we have huge next apps whose files have many imports. The more imports a file has, the more chunk IDs exist in each file, which is more likely that the order of the arrays is incorrect - this was easy to notice.

## Expected

* chunks should be the same when the build doesn't change

## Proposed solution

* sort the webpack chunk ID array so that the array stays the same if the contents are the same

(not sure whether this needs to come from the nextjs side or the webpack side, but if someone points me in the right direction, i'll try to open a PR)

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 20.12.0
  npm: 10.5.0
  Yarn: 4.4.0
  pnpm: N/A
Relevant Packages:
  next: 14.2.14 // Latest available version is detected (14.2.14).
  eslint-config-next: 14.2.14
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.6.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Module Resolution, Output (export/standalone), Performance, Webpack

### Which stage(s) are affected? (Select all that apply)

next build (local), Vercel (Deployed)

**Comment by unstubbable** [comment]
This may be fixed with #66546, which is concerned with server bundles and not client chunks, but sorting the entrypoints may possibly fix both issues. Needs to be investigated.

**Comment by chriskrogh** [comment]
> This may be fixed with #66546, which is concerned with server bundles and not client chunks, but sorting the entrypoints may possibly fix both issues. Needs to be investigated.

@unstubbable think sorting chunk ids might need to be done from the webpack side. will look into this.
